sdk/maps/azure-maps-search/src/test/java/com/azure/maps/search/MapsSearchAsyncClientTest.java

```diff
@@ -0,0 +1,722 @@
+        SyncPoller<BatchSearchResult, BatchSearchResult> syncPoller = setPollInterval(pollerFlux.getSyncPoller());
+        syncPoller.waitForCompletion();
+        BatchSearchResult actualResult =  syncPoller.getFinalResult();
+        BatchSearchResult expectedResult = TestUtils.getExpectedBeginFuzzySearchBatch();
+        validateBeginFuzzySearchBatch(expectedResult, actualResult);
+     }
+
+    // Test async begin search address batch
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.maps.search.TestUtils#getTestParameters")
+    public void testAsyncBeginSearchAddressBatch(HttpClient httpClient, MapsSearchServiceVersion serviceVersion) throws IOException {
+        client = getMapsSearchAsyncClient(httpClient, serviceVersion);
+        List<SearchAddressOptions> searchAddressOptionsList = new ArrayList<>();
+        searchAddressOptionsList.add(new SearchAddressOptions("400 Broad St, Seattle, WA 98109").setTop(3));
+        searchAddressOptionsList.add(new SearchAddressOptions("One, Microsoft Way, Redmond, WA 98052").setTop(3));
+        searchAddressOptionsList.add(new SearchAddressOptions("350 5th Ave, New York, NY 10118").setTop(3));
+        searchAddressOptionsList.add(new SearchAddressOptions("1 Main Street"));
+        PollerFlux<BatchSearchResult, BatchSearchResult> pollerFlux = client.beginSearchAddressBatch(searchAddressOptionsList);
+        SyncPoller<BatchSearchResult, BatchSearchResult> syncPoller = setPollInterval(pollerFlux.getSyncPoller());
+        syncPoller.waitForCompletion();
+        BatchSearchResult actualResult = syncPoller.getFinalResult();
+        BatchSearchResult expectedResult = TestUtils.getExpectedBeginSearchAddressBatch();
+        validateBeginSearchAddressBatch(expectedResult, actualResult);
+    }
+
+    // Test async begin reverse search address batch
+    
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.maps.search.TestUtils#getTestParameters")
+    public void testAsyncBeginReverSearchAddressBatch(HttpClient httpClient, MapsSearchServiceVersion serviceVersion) throws IOException {
+        client = getMapsSearchAsyncClient(httpClient, serviceVersion);
+        List<ReverseSearchAddressOptions> reverseOptionsList = new ArrayList<>();
+        reverseOptionsList.add(new ReverseSearchAddressOptions(new LatLong(48.858561, 2.294911)));
+        reverseOptionsList.add(new ReverseSearchAddressOptions(new LatLong(47.639765, -122.127896)).setRadiusInMeters(5000));
+        reverseOptionsList.add(new ReverseSearchAddressOptions(new LatLong(47.621028, -122.348170)));
+        PollerFlux<BatchReverseSearchResult, BatchReverseSearchResult> pollerFlux = client.beginReverseSearchAddressBatch(reverseOptionsList);
+        SyncPoller<BatchReverseSearchResult, BatchReverseSearchResult> syncPoller = setPollInterval(pollerFlux.getSyncPoller());
+        syncPoller.waitForCompletion();
```

Comment:
I don't think you need waitForCompletion if you just call getFinalResult(). you can test without it.